pkg/util/push/push.go

```diff
@@ -33,12 +34,27 @@ type bufHolder struct {
 	buf []byte
 }
 
+// TODO: the idea of this interface is to represent the error interface of
+// push.Func, which would then have the following signature:
+// type Fund func(ctx context.Context, req *Request) Error.
```

Comment:
I don't think that `push.Func` has to return a `push.Error`. There are a lot of errors that can happen which aren't something specific to `push.Func`.

I don't think we need this interface at all.

Also, nit:
```suggestion
// type Func func(ctx context.Context, req *Request) Error.
```

Reply:
> I don't think we need this interface at all.

Agree.

Reply:
This interface has been removed.